### DISCUSSION TITLE
NVFlare with Basic auth

### DISCUSSION THREAD

**Discussion by nfmanasd28** [NVIDIA/NVFlare#1875] [Q&A]
Hi, 
We need a setup where there is HAPROXY between the server and the client. To support this,  we made some changes in the poc mode to support authentication via HAPROXY. 
[NVFlare Basic auth in POC MODE](https://github.com/NVIDIA/NVFlare/compare/dev...harishnference:NVFlare:harish_nvflare)

We have also tried a similar setup in Provisioning mode 
[Provisioning mode via HPAROXY](https://github.com/NVIDIA/NVFlare/discussions/1871)

Is there a existing method of what we are trying to achieve?

**Comment by chesterxgchen** [comment]
May I first try to understand what you try to achieve. 

what I understand is the following: 

1. There is an reverse proxy between client and server.  you would like to sever and client be able to communicate between them pass-through the reverse proxy

2. The reverse proxy is using HAPROXY ?  WHY HAPROXY, can you use nginx ? ( I seen people successfully using client and server with muliple ngnix as reverse proxies in between)

3. not sure authentication part, do you need authentication ? 
    if yes,  sounds like you need to authenticate to pass through HAProxy ? extra security  need ?

**Reply by nfmanasd28** [reply]
1. Correct. I would like to have HAPROXY as a reverse proxy between client and the server
2. We use HAPROXY as a part of our system configuration. We would prefer the reverse to be HAPROXY.
3. Yes, we need authentication. Besides MTLS, (that nvflare uses by default in provisioning mode), is there a support to use tls or basic auth?

**Comment by chesterxgchen** [comment]
@kkersten @nvidianz @IsaacYangSLA  can you help with this ?

**Comment by chesterxgchen** [comment]
the offline call addressed this issue. I am close this discussion
